OpenRA.Mods.Common/EditorBrushes/EditorTileBrush.cs

```diff
@@ -106,8 +124,80 @@ public bool HandleMouseInput(MouseInput mi)
 					}
 				}
 			}
+		}
 
-			return true;
+		void FloodFillWithBrush(CPos cell, bool isMoving)
+		{
+			var map = world.Map;
+			var mapTiles = map.MapTiles.Value;
+			var replace = mapTiles[cell];
+
+			if (replace.Type == Template)
+				return;
+
+			var queue = new Queue<CPos>();
+			var touched = new bool[map.MapSize.X, map.MapSize.Y];
+
+			var rules = map.Rules;
+			var tileset = rules.TileSets[map.Tileset];
+			var template = tileset.Templates[Template];
+
+			Action<CPos> maybeEnqueue = (newCell) =>
+			{
+				if (map.Contains(cell) && !touched[newCell.X, newCell.Y])
```

Comment:
Indexing a 2D array with cell coordinates unfortunately isn't a valid thing to do.  This will crash the TS mod.  Use a `CellLayer<bool>` instead.